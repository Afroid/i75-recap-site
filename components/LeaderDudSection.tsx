import React from "react";
import ReactMarkdown from "react-markdown";
import { LDSSection as LDS } from "@/types/types";


export default function LeaderDudSection({ data }: { data: LDS }) {
  return (
    <div>
      {/* Points Leader */}
      <h3 className="text-xl font-semibold mb-1">
        Points Leader: {data.pointsLeader.name}
        <span className="ml-1 text-green-500">
          ({data.pointsLeader.points})
        </span>
      </h3>
      {data.pointsLeader.notes && (
        <ReactMarkdown className="text-gray-800 mb-4 whitespace-pre-wrap">
          {data.pointsLeader.notes}
        </ReactMarkdown>
      )}

      {/* Dud */}
      <h3 className="text-xl font-semibold mb-1">
        Dud: {data.dud.name}
        <span className="ml-1 text-red-500">
          ({data.dud.points})
        </span>
      </h3>
      {data.dud.notes && (
        <ReactMarkdown className="text-gray-800 whitespace-pre-wrap">
          {data.dud.notes}
        </ReactMarkdown>
      )}
    </div>

  );
}
