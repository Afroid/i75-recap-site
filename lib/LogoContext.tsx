"use client";
import { createContext, useContext, useMemo } from "react";

const logos = [
  "/default-logo.png",
  "/default-logo-alt.png",
  "/default-RWB.png",
];

const LogoContext = createContext<string | null>(null);

export function LogoProvider({ children }: { children: React.ReactNode }) {
  // Randomizes only once on initial render
  const logo = useMemo(
    () => logos[Math.floor(Math.random() * logos.length)],
    []
  );
  return (
    <LogoContext.Provider value={logo}>
      {children}
    </LogoContext.Provider>
  );
}

export function useLogo() {
  const logo = useContext(LogoContext);
  if (logo === null) throw new Error("useLogo must be inside LogoProvider");
  return logo;
}
