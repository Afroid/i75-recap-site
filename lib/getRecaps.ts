import recap2024 from "@/data/recaps/2024.json";
import recap2023 from "@/data/recaps/2023.json";
import { RecapData } from "@/types/types";

export function getRecaps(): Record<number, number[]> {
  const allData: RecapData[] = [recap2024, recap2023]; // Add more as needed
  const summary: Record<number, number[]> = {};

  for (const season of allData) {
    summary[season.year] = season.recaps.map((r) => r.week);
  }

  return summary;
}
