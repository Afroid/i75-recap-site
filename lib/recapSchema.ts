import { z } from "zod";

const RecapMatchupSchema = z.object({
  team1: z.string(),
  team2: z.string(),
  winner: z.string(),
  score: z.string(),
  breakdown: z.string(),
  imageUrl: z.string().optional(),
  imageAlt: z.string().optional(),
});

const GameNotesSectionSchema = z.object({
  type: z.literal("gameNotes"),
  pointsLeader: z.object({
    name: z.string(),
    points: z.number(),
    notes: z.string().optional(),
  }),
  dud: z.object({
    name: z.string(),
    points: z.number(),
    notes: z.string().optional(),
  }),
  matchups: z.array(RecapMatchupSchema),
});

const RecapSectionSchema = z.discriminatedUnion("type", [
  z.object({
    type: z.literal("intro"),
    content: z.string(),
  }),
  z.object({
    type: z.literal("tidbits"),
    intro: z.string().optional(),
    bullets: z.array(z.string()).optional(),
    outro: z.string().optional(),
    imageUrl: z.string().optional(),
    imageAlt: z.string().optional(),
  }),
  z.object({
    type: z.literal("injuries"),
    intro: z.string().optional(),
    bullets: z.array(z.string()).optional(),
    outro: z.string().optional(),
    imageUrl: z.string().optional(),
    imageAlt: z.string().optional(),
  }),
  z.object({
    type: z.literal("sidePieces"),
    intro: z.string().optional(),
    bullets: z.array(z.string()).optional(),
    outro: z.string().optional(),
    imageUrl: z.string().optional(),
    imageAlt: z.string().optional(),
  }),
  z.object({
    type: z.literal("matchups"),
    intro: z.string().optional(),
    bullets: z.array(z.string()).optional(),
    outro: z.string().optional(),
    imageUrl: z.string().optional(),
    imageAlt: z.string().optional(),
  }),
  z.object({
    type: z.literal("farewell"),
    intro: z.string().optional(),
    bullets: z.array(z.string()).optional(),
    outro: z.string().optional(),
    imageUrl: z.string().optional(),
    imageAlt: z.string().optional(),
  })
]);

const RecapWeekSchema = z.object({
  week: z.number(),
  title: z.string(),
  sections: z.array(z.union([RecapSectionSchema, GameNotesSectionSchema])),
});

export const RecapDataSchema = z.object({
  year: z.number(),
  recaps: z.array(RecapWeekSchema),
});
