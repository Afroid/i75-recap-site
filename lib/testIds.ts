export const TestIds = {
  /**
   * Pages
   */

  // Index - App Level
  SITE_HEADER: "site-header",
  MAIN_CONTENT: "main-content",
  PAGE_WRAPPER: "page-wrapper",

  // Home Page
  HOME_MAIN:   "home-main",
  HOME_HERO_LOGO: "home-hero-logo",
  HOME_INTRO_SECTION: "home-intro-section",
  HOME_HERO_HEADING: "home-hero-heading",
  HOME_CTA_CONTAINER: "home-cta-container",
  HOME_VIEW_ALL_RECAPS_BUTTON: "home-view-all-recaps-button",
  HOME_LEARN_MORE_BUTTON: "home-learn-more-button",

  // 404 Page
  NOT_FOUND_MAIN: "not-found-main",
  RETURN_HOME_LINK: "return-home-link",

  // About Page
  ABOUT_MAIN: "about-main",

  // Contact Page
  CONTACT_MAIN: "contact-main",
  CONTACT_EMAIL_LINK: "contact-email-link",
  CONTACT_FORM: "contact-form",
  CONTACT_FORM_NAME_SECTION: "contact-form-name-section",
  CONTACT_FORM_EMAIL_SECTION: "contact-form-email-section",
  CONTACT_FORM_MESSAGE_SECTION: "contact-form-message-section",
  CONTACT_FORM_SUBMIT: "contact-form-submit",

  /**
   * Recap Pages - (pages/recaps)
   */

  // Recap Page

  // View All Recaps Page
  VIEW_ALL_RECAPS: "view-all-recaps",
  // Year Page

  /**
   * Components
   */

  // Breadcrumb Component
  BREADCRUMB: "breadcrumb",
  BREADCRUMB_NAV: "breadcrumb-nav",
  BREADCRUMB_LIST: "breadcrumb-list",
  BREADCRUMB_ITEM: "breadcrumb-item",
  BREADRCRUMB_LINK: "breadcrumb-link",
  BREADCRUMB_CURRENT: "breadcrumb-current",
  BREADCRUMB_SEPARATOR: "breadcrumb-separator",

  // DesktopNavbar Component

  // GameNotesSection.tsx
  GAME_NOTES_SECTION: "game-notes-section",
  GAME_NOTES_ITEM: "game-notes-item",
  GAME_NOTES_SCOREBOARD: "game-notes-scoreboard",
  GAME_NOTES_BREAKDOWN: "game-notes-breakdown",
  GAME_NOTES_IMAGE_CONTAINER: "game-notes-image-container",
  GAME_NOTES_IMAGE: "game-notes-image",

  // Header.tsx
  HEADER_SITE_HEADER: "site-header",
  HEADER_MOBILE_HEADER: "mobile-header",
  HEADER_HAMBURGER_BUTTON: "header-hamburger-button",
  HEADER_MOBILE_LOGO: "header-mobile-logo",
  HEADER_DESKTOP_LOGO: "header-desktop-logo",
  HEADER_DESKTOP_NAV: "header-desktop-nav",
  HEADER_MOBILE_DRAWER: "header-mobile-drawer",

  // HeaderLogo.tsx
  HEADER_LOGO_LINK: "header-logo-link",
  HEADER_LOGO_BG_DARK: "header-logo-bg-dark",
  HEADER_LOGO_BG_LIGHT: "header-logo-bg-light",
  HEADER_LOGO_CONTAINER: "header-logo-container",
  HEADER_LOGO_NAME: "header-logo-name",
  HEADER_LOGO_DYNAMIC: "header-logo-dynamic",

  // LeaderDudSection.tsx
  LDS_SECTION: "lds-section",
  LDS_LEADER_HEADING: "lds-leader-heading",
  LDS_LEADER_POINTS: "lds-leader-points",
  LDS_LEADER_NOTES: "lds-leader-notes",
  LDS_DUD_HEADING: "lds-dud-heading",
  LDS_DUD_POINTS: "lds-dud-points",
  LDS_DUD_NOTES: "lds-dud-notes",

  // MobileDrawer.tsx
  MOBILE_SEASON_TOGGLE: "mobile-season-toggle",
  MOBILE_SEASON_WEEKS: "mobile-season-weeks",
  MOBILE_WEEK_LINK: "mobile-week-link",
  MOBILE_DRAWER_OVERLAY: "mobile-drawer-overlay",
  MOBILE_DRAWER: "mobile-drawer",
  MOBILE_DRAWER_TITLE: "mobile-drawer-title",
  MOBILE_OLDER_SEASON_TITLE: "mobile-older-season-title",
  MOBILE_TOGGLE_OLDER_SEASONS: "mobile-toggle-older-seasons",

  // RecapLayout.tsx
  RECAP_TITLE: "recap-title",
  RECAP_SECTION: "recap-section",
  RECAP_HEADING: "recap-heading",
  RECAP_INTRO: "recap-intro",
  RECAP_BULLETS: "recap-bullets",
  RECAP_BULLET: "recap-bullet",
  RECAP_CONTENT: "recap-content",
  RECAP_IMAGE: "recap-image",

  // Scoreboard.tsx
  SCOREBOARD_CONTAINER: "scoreboard-container",
  SCOREBOARD_ENTRY: "scoreboard-entry",
  SCOREBOARD_LOGO: "scoreboard-logo",
  SCOREBOARD_NAME: "scoreboard-name",
  SCOREBOARD_RECORD: "scoreboard-record",
  SCOREBOARD_SCORE: "scoreboard-score",

  // TeamLogo.tsx
  TEAM_LOGO_IMAGE: "team-logo-image",

  // Static Links
  NAV_LINK_HOME: "nav-link-home",
  NAV_LINK_ABOUT: "nav-link-about",
  NAV_LINK_CONTACT: "nav-link-contact"
}
