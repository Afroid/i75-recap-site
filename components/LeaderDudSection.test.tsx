import React from "react";
import { render, screen } from "@testing-library/react";
import { TestIds } from "@/lib/testIds";

// Stub react-markdown so it just renders its children
jest.mock("react-markdown", () => ({
  __esModule: true,
  default: ({
    children,
    className,
    "data-testid": testId,
  }: {
    children: React.ReactNode;
    className?: string;
    "data-testid"?: string;
  }) => (
    <div className={className} data-testid={testId}>
      {children}
    </div>
  ),
}));

/**
 * Importing after the mocks ensures the real copies of those dependencies don't get loaded.
 * This ensures any imports from next/head or whatever get routed to the stubbed versions, rather
 * than real implementation. This makes sure jest.mock calls will have effect.
 */
// Now import LeaderDudSection (after the mocks)
import LeaderDudSection from "./LeaderDudSection";
import type { LDSSection as LDS } from "@/types/types";

describe("LeaderDudSection", () => {
  it("renders leader and dud headings with points, no notes by default", () => {
    const data: LDS = {
      type: "leaderDud",
      pointsLeader: {
        name: "Player X",
        points: 42
      },
      dud: { name: "Player Z", points: 7 },
    };

    render(<LeaderDudSection data={data} />);

    // Headings and points
    expect(screen.getByTestId(TestIds.LDS_LEADER_HEADING)).toHaveTextContent(
      "Points Leader: Player X"
    );
    expect(screen.getByTestId(TestIds.LDS_LEADER_POINTS)).toHaveTextContent("(42)");

    expect(screen.getByTestId(TestIds.LDS_DUD_HEADING)).toHaveTextContent("Dud: Player Z");
    expect(screen.getByTestId(TestIds.LDS_DUD_POINTS)).toHaveTextContent("(7)");

    // Notes should not be rendered
    expect(screen.queryByTestId(TestIds.LDS_DUD_NOTES)).toBeNull();
    expect(screen.queryByTestId(TestIds.LDS_DUD_NOTES)).toBeNull();
  });

  it("renders notes when provided", () => {
    const data: LDS = {
      type: "leaderDud",
      pointsLeader: {
        name: "Player X",
        points: 42,
        notes: "Great game!",
      },
      dud: {
        name: "Player Z",
        points: 7,
        notes: "Could have tried harder.",
      },
    };

    render(<LeaderDudSection data={data} />);

    expect(screen.getByTestId(TestIds.LDS_LEADER_NOTES)).toHaveTextContent(
      "Great game!"
    );
    expect(screen.getByTestId(TestIds.LDS_DUD_NOTES)).toHaveTextContent(
      "Could have tried harder."
    );
  });

  it("matches snapshot when leader has notes but dud does not", () => {
    const data: LDS = {
      type: "leaderDud",
      pointsLeader: {
        name: "Player X",
        points: 42,
        notes: "MVP of the week",
      },
      dud: { name: "Player Z", points: 7 },
    };

    const { asFragment } = render(<LeaderDudSection data={data} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
