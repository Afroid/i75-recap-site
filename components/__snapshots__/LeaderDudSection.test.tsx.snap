// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaderDudSection matches snapshot when leader has notes but dud does not 1`] = `
<DocumentFragment>
  <div
    data-testid="lds-section"
  >
    <h3
      class="text-xl font-semibold mb-1"
      data-testid="lds-leader-heading"
    >
      Points Leader: Player X
      <span
        class="ml-1 text-green-500"
        data-testid="lds-leader-points"
      >
        (42)
      </span>
    </h3>
    <div
      class="text-gray-800 mb-4 whitespace-pre-wrap"
      data-testid="lds-leader-notes"
    >
      MVP of the week
    </div>
    <h3
      class="text-xl font-semibold mb-1"
      data-testid="lds-dud-heading"
    >
      Dud: Player Z
      <span
        class="ml-1 text-red-500"
        data-testid="lds-dud-points"
      >
        (7)
      </span>
    </h3>
  </div>
</DocumentFragment>
`;
